index.js

```diff
@@ -101,14 +101,21 @@ function csurf (options) {
       return token
     }
 
+    var submitted = value(req)
+
     // generate & set secret
     if (!secret) {
+      if (!ignoreMethod[req.method] && submitted) {
+        return next(createError(403, 'csrf token submitted but there is no active session', {
```

Comment:
Makes a lot of sense, you have of course a much better understanding of csurf while I just know the *session* side of it. Will fix it.